pkgs/os-specific/linux/syslinux/default.nix

```diff
@@ -65,6 +67,8 @@ stdenv.mkDerivation {
       "0018-prevent-pow-optimization.patch"
       "26f0e7b2"
       "sha256-dVzXBi/oSV9vYgU85mRFHBKuZdup+1x1BipJX74ED7E=")
+    # Fixes build with "modern" gnu-efi
+    ./import-efisetjmp.patch
```

Comment:
The parent of the fix was already `b404870`

 - https://github.com/samueldr/syslinux/commit/68defee52f4eba82eefaeea17f21c7498448dd6b

Aren't we supposed to prefer `fetchpatch` to dumping patches in Nixpkgs? (TBF I don't know what is the current recommended way to handle this, if there is any...)

Reply:
Yes, usually but since you mirrored the repo from or.cz to github, we cannot rely on the main project keeping the hash. If you delete the repo by accident, it's gone. I think the same cannot happen for open PRs.

Reply:
> Aren't we supposed to prefer `fetchpatch` to dumping patches in Nixpkgs? (TBF I don't know what is the current recommended way to handle this, if there is any...)

I originally referred to your commit but pulled it in on @SuperSandro2000's request above :) .

From some recent chat in the 'security discussion' matrix channel (sorry I don't have a reference right now) my impression was that we prefer `fetchpatch` for changes that have been merged, but for PRs it seems like more of a grey area: if the PR is closed, squashed or rebased the commit might eventually get garbage collected. Since this was a fairly trivial change I figured it makes sense to just embed it.

I agree it would be good to have clearer recommendations on this in `CONTRIBUTING.md`.

Reply:
Yes, sorry to put this on "our" PR, but I actually was curious about the current recommendations. My recollections were to not vendor patches. (Though personally, I think all patches not produced by upstreams should be vendored for increased reliability and discoverability...)

Reply:
opened #317314 (but let's not let that hold back this PR)